app/src/ui/banners/successful-cherry-pick.tsx

```diff
@@ -1,32 +1,45 @@
 import * as React from 'react'
+import { LinkButton } from '../lib/link-button'
 import { Octicon, OcticonSymbol } from '../octicons'
 import { Banner } from './banner'
 
-export function SuccessfulCherryPick({
-  targetBranchName,
-  countCherryPicked,
-  onDismissed,
-}: {
+interface ISuccessfulCherryPickBannerProps {
   readonly targetBranchName: string
   readonly countCherryPicked: number
   readonly onDismissed: () => void
-}) {
-  const pluralized = countCherryPicked === 1 ? 'commit' : 'commits'
-  return (
-    <Banner
-      id="successful-cherry-pick"
-      timeout={7500}
-      onDismissed={onDismissed}
-    >
-      <div className="green-circle">
-        <Octicon className="check-icon" symbol={OcticonSymbol.check} />
-      </div>
-      <div className="banner-message">
-        <span>
-          Successfully copied {countCherryPicked} {pluralized} to{' '}
-          <strong>{targetBranchName}</strong>.
-        </span>
-      </div>
-    </Banner>
-  )
+  readonly onUndoCherryPick: () => void
+}
+
+export class SuccessfulCherryPick extends React.Component<
+  ISuccessfulCherryPickBannerProps,
+  {}
+> {
+  private undo = () => {
+    this.props.onDismissed()
+    this.props.onUndoCherryPick()
+  }
+
+  public render() {
+    const { countCherryPicked, onDismissed, targetBranchName } = this.props
+
+    const pluralized = countCherryPicked === 1 ? 'commit' : 'commits'
+    return (
+      <Banner
+        id="successful-cherry-pick"
+        timeout={7500}
+        onDismissed={onDismissed}
+      >
+        <div className="green-circle">
+          <Octicon className="check-icon" symbol={OcticonSymbol.check} />
+        </div>
+        <div className="banner-message">
+          <span>
+            Successfully copied {countCherryPicked} {pluralized} to{' '}
+            <strong>{targetBranchName}</strong>.{' '}
+            <LinkButton onClick={this.undo}>Undo</LinkButton>
```

Comment:
If we can't get the original sha for scenarios where cherry-picking was initiated outside of our app, we should probably remove this link, right?

Reply:
(see answer to other comment)